core/kernel/user_ta.c

```diff
@@ -489,12 +489,18 @@ TEE_Result tee_ta_init_user_ta_session(const TEE_UUID *uuid,
 		return res;
 	}
 
-	utc->ta_ctx.is_initializing = true;
-
 #ifdef CFG_TA_PAUTH
-	crypto_rng_read(&utc->uctx.keys, sizeof(utc->uctx.keys));
+	res = crypto_rng_read(&utc->uctx.keys, sizeof(utc->uctx.keys));
```

Comment:
How about moving this to right after `calloc()` to simplify the cleanup action?

Reply:
okay